src/state/types.ts

```diff
@@ -195,12 +195,12 @@ export interface Market {
 }
 
 export interface Bet {
-  id: string
-  hash: string
+  id?: string
+  hash?: string
   amount: number
   position: BetPosition
   claimed: boolean
-  user: PredictionUser
+  user?: PredictionUser
```

Comment:
This is in place of using `Partial<Bet>` which was causing all sorts of problems.